src/applications/personalization/profile-2/components/MobileMenuTrigger.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState, useRef } from 'react';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+
+import {
+  closeSideNav as closeSideNavAction,
+  openSideNav as openSideNavAction,
+  pinMenuTrigger as pinMenuTriggerAction,
+  unpinMenuTrigger as unpinMenuTriggerAction,
+} from '../actions';
+
+const MobileMenuTrigger = ({
+  closeSideNav,
+  focusTriggerButton,
+  openSideNav,
+  isMenuTriggerPinned,
+  pinMenuTrigger,
+  unpinMenuTrigger,
+}) => {
+  const button = useRef(null);
+  const placeholder = useRef(null);
+
+  const [mobileBreakpoint] = useState(767);
+  const [isMobile, setIsMobile] = useState(false);
+  const [triggerPosition, setTriggerPosition] = useState(null);
+  const [triggerHeight, setTriggerHeight] = useState(null);
+
+  // init some local state
+  useEffect(
+    () => {
+      setTriggerHeight(window.getComputedStyle(button.current).height);
+      setTriggerPosition(placeholder.current.offsetTop);
+      setIsMobile(window.innerWidth <= mobileBreakpoint);
+    },
+    [mobileBreakpoint],
+  );
+
+  useEffect(() => {
+    const handleResize = () => {
+      const mobile = window.innerWidth <= mobileBreakpoint;
+      if (!mobile) {
+        if (isMobile) {
```

Comment:
This little part is very confusing :)

Reply:
I'll take another pass at the variable names and/or add some comments to add clarity. Effectively all this is trying to accomplish is "if the window just shrunk down to mobile size or smaller, close the side nav"